src/ScaleUnitManagement/ScaleUnitFeatureManager/Utilities/IISAdministrationHelper.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft;
+using Microsoft.Web.Administration;
+using ScaleUnitManagement.Utilities;
+
+namespace ScaleUnitManagement.ScaleUnitFeatureManager.Utilities
+{
+    internal static class IISAdministrationHelper
+    {
+        internal static void CreateSite(string siteName, string siteRoot,
+            string bindingInformation, string certSubject, string appPoolName)
+        {
+            Requires.NotNull(siteName, nameof(siteName));
+            Requires.NotNull(siteRoot, nameof(siteRoot));
+            Requires.NotNull(bindingInformation, nameof(bindingInformation));
+            Requires.NotNull(certSubject, nameof(certSubject));
+            Requires.NotNull(appPoolName, nameof(appPoolName));
+
+            CreateAppPool(appPoolName);
+
+            using (ServerManager manager = new ServerManager())
+            {
+                Site site = manager.Sites.FirstOrDefault((s) => s.Name.Equals(siteName));
+
+                if (site != null)
+                {
+                    manager.Sites.Remove(site);
+                }
+
+                site = manager.Sites.Add(siteName, siteRoot, 443);
+                site.Applications[0].ApplicationPoolName = appPoolName;
+
+                site.Bindings.Clear();
+                site.Bindings.Add(bindingInformation,
+                    CertificateStoreHelper.GetCertificateFromLocalMachineStore(certSubject),
+                    CertificateStoreHelper.PersonalCertificateStoreName);
+
+                manager.CommitChanges();
+            }
+        }
+
+        private static void CreateAppPool(string appPoolName)
+        {
+            // we are assuming HubAppPool is already and doesnt need changes. 
```

Comment:
can we handle the hub app pool the same way as with the scale unit app pool? if the app pool exists we remove it and we replace it with a new one.